### ISSUE TITLE
API Proposal

### ISSUE THREAD

**Issue by kettanaito** [kettanaito/atomic-layout#116]
This ticket is going to list API proposals for the upcoming major version. Feel free to vote on the proposals that make more sense to you.

**Comment by kettanaito** [comment]
# `Composition.props.areas`

Replace the current `template` prop with the `area` prop.

## Motivation

* Currently `template` prop aliases `grid-template-areas`, but not `grid-template`. This creates confusion and makes the usage not intuitive.
* `areas` prop is shorter, without sacrificing the readability.

## Example

```diff
- <Composition template="header footer" templateMd="header footer" />
+ <Composition areas="header footer" areasMd="header footer" />
```

**Comment by redraushan** [comment]
Hi @kettanaito,

Are we first going to deprecate `template` props and make it work together with `areas` or it will be silently replaced `areas` with breaking changes?

**Comment by kettanaito** [comment]
@redraushan, I believe all the proposals will be adapted as a non-breaking changes, and the old API will be removed in the version `1.0`.

Regarding the `template` prop, I think it would make sense to use it as a `grid-template` instead. I'm conducting a proof of concept how to parse the areas when they include row/column sizing.

**Comment by kettanaito** [comment]
# Deprecate `Layout.getBreakpoint`

Deprecate and remove `getBreakpoint` method of the `Layout` interface.

## Motivation

Once layout options are stored on the root level (#153) it becomes easy to reference breakpoints directly, makng an explicit API redundant.

```jsx
import Layout from 'atomic-layout'

Layout.breakpoints.md.minWidth
```

**Comment by kettanaito** [comment]
# Deprecate `Layout.getBreakpointNames`

Remove the respective method from the `Layout` interface.

## Motivation

With #153 implemented, you can get the list of breakpoint names directly, using `Object.keys` on the layout breakpoints:

```jsx
import Layout from 'atomic-layout'

const breakpointNames = Object.keys(Layout.breakpoints)
```

**Comment by kettanaito** [comment]
Implemented in #183.
